java/src/org/openqa/selenium/remote/ErrorHandler.java

```diff
@@ -110,7 +110,9 @@ public Response throwIfResponseFailed(Response response, long duration) throws R
       if (!rawErrorData.containsKey(MESSAGE) && rawErrorData.containsKey("value")) {
         try {
           rawErrorData = (Map<String, Object>) rawErrorData.get("value");
-        } catch (ClassCastException cce) {}
+        } catch (ClassCastException cce) {
+          throw new RuntimeException(cce);
+        }
```

Comment:
I don't think this is ideal because if this happens in Grid, the client side won't be able to see the error.

Probably the sane thing is to do the same that is being done in line 119:

```
message = String.valueOf(e);
```

Reply:
@diemol I've added assigning to `message` and resolved conflicts with `trunk`